server/app/services/question/types/MultiOptionQuestionDefinitionConfig.java

```diff
@@ -0,0 +1,88 @@
+import java.time.Instant;
+import java.util.Optional;
+import java.util.OptionalLong;
+import services.LocalizedStrings;
+import services.question.QuestionOption;
+import services.question.types.MultiOptionQuestionDefinition.MultiOptionQuestionType;
+import services.question.types.MultiOptionQuestionDefinition.MultiOptionValidationPredicates;
+
+@AutoValue
+public abstract class MultiOptionQuestionDefinitionConfig {
+  abstract OptionalLong id();
+
+  abstract String name();
+
+  abstract Optional<Long> enumeratorId();
+
+  abstract String description();
+
+  abstract LocalizedStrings questionText();
+
+  abstract LocalizedStrings questionHelpText();
+
+  abstract ImmutableList<QuestionOption> questionOptions();
+
+  abstract Optional<MultiOptionValidationPredicates> validationPredicates();
+
+  abstract Optional<Instant> lastModifiedTime();
+
+  abstract MultiOptionQuestionType multiOptionQuestionType();
+
+  public static RequiredMultiOptionQuestionType builder() {
+    return new AutoValue_MultiOptionQuestionDefinitionConfig.Builder();
+  }
+
+  static final MultiOptionValidationPredicates SINGLE_SELECT_PREDICATE =
+      MultiOptionValidationPredicates.create(1, 1);
+
+  public interface RequiredMultiOptionQuestionType {
+    Builder setMultiOptionQuestionType(MultiOptionQuestionType multiOptionQuestionType);
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder implements RequiredMultiOptionQuestionType {
+    public abstract Builder setId(long id);
+
+    public abstract Builder setId(OptionalLong id);
+
+    public abstract Builder setName(String name);
+
+    public abstract Builder setEnumeratorId(long enumeratorId);
+
+    public abstract Builder setEnumeratorId(Optional<Long> enumeratorId);
+
+    public abstract Builder setDescription(String description);
+
+    public abstract Builder setQuestionText(LocalizedStrings questionText);
+
+    public abstract Builder setQuestionHelpText(LocalizedStrings questionHelpText);
+
+    public abstract Builder setQuestionOptions(ImmutableList<QuestionOption> questionOptions);
+
+    public abstract Builder setValidationPredicates(
+        MultiOptionValidationPredicates validationPredicates);
+
+    public abstract Builder setLastModifiedTime(Instant lastModifiedTime);
+
+    public abstract Builder setLastModifiedTime(Optional<Instant> lastModifiedTime);
+
+    abstract MultiOptionQuestionType multiOptionQuestionType();
+
+    abstract MultiOptionQuestionDefinitionConfig autoBuild();
+
+    public MultiOptionQuestionDefinitionConfig build() {
+      if (multiOptionQuestionType() == MultiOptionQuestionType.DROPDOWN
+          || multiOptionQuestionType() == MultiOptionQuestionType.CHECKBOX) {
+        // If we are using a dropdown or checkbox, set the SINGLE_SELECT_PREDICATE to ensure only
+        // one selection can be made.
+        setValidationPredicates(SINGLE_SELECT_PREDICATE);
+      }
+
+      return autoBuild();
+    }
+  }
```

Comment:
The `build()` method above replaces all the custom logic in `CheckboxQuestionDefinition` and `DropdownQuestionDefinition` in just a few lines; namely, that those types must have the `SINGLE_SELECT_PREDICATE`.

Reply:
So much better!!